File: paper/origin/src/main/java/com/github/theword/queqiao/command/CommandExecutor.java
package com.github.theword.queqiao.command;

import com.github.theword.queqiao.tool.GlobalContext;
import com.github.theword.queqiao.tool.command.CommandExecutorHelper;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabExecutor;
import org.jetbrains.annotations.NotNull;

import java.util.List;


public class CommandExecutor implements TabExecutor {
    private final CommandExecutorHelper commandExecutorHelper;

    public CommandExecutor() {
        this.commandExecutorHelper = new CommandExecutorHelper();
    }

    @Override
    public boolean onCommand(@NotNull CommandSender commandSender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        try {
            commandExecutorHelper.execute(commandSender, args);
            return true;
        } catch (Exception e) {
            GlobalContext.getHandleCommandReturnMessageService().sendReturnMessage(command, "执行命令时出现异常！");
            GlobalContext.getLogger().error(e.getMessage());
            return false;
        }
    }

    @Override
    public List<String> onTabComplete(@NotNull CommandSender commandSender, @NotNull Command command, @NotNull String s, @NotNull String[] args) {
        return commandExecutorHelper.tabComplete(commandSender, args);
    }
}

File: paper/origin/src/main/kotlin/com/github/theword/queqiao/command/CommandExecutor.kt
package com.github.theword.queqiao.command

import com.github.theword.queqiao.tool.command.CommandExecutorHelper
import com.github.theword.queqiao.tool.constant.CommandConstant
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.command.TabExecutor

class CommandExecutor : TabExecutor {

    private val commandExecutorHelper = CommandExecutorHelper()

    override fun onCommand(
        commandSender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>
    ): Boolean {
        val result = commandExecutorHelper.execute(commandSender, args)
        return result == CommandConstant.SUCCESS_SIGNAL
    }

    override fun onTabComplete(
        commandSender: CommandSender,
        command: Command,
        alias: String,
        args: Array<String>
    ): List<String>? {
        return commandExecutorHelper.tabComplete(commandSender, args)
    }
}